### PR TITLE
ui: Customize diagnose duration

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <me@breeswish.org>

### CODE REVIEW

File: ui/src/apps/diagnose/components/DiagnoseGenerator.tsx [hunk 1]
```diff
@@ -52,18 +64,49 @@ export default function DiagnoseGenerator() {
...
-          label={t('diagnose.generate.range_duration')}
-        >
-          <Select style={{ width: 120 }}>
-            <Select.Option value={5}>5 min</Select.Option>
-            <Select.Option value={10}>10 min</Select.Option>
-            <Select.Option value={30}>30 min</Select.Option>
-            <Select.Option value={60}>1 hour</Select.Option>
-            <Select.Option value={24 * 60}>1 day</Select.Option>
-          </Select>
+        <Form.Item label={t('diagnose.generate.range_duration')}>
```
Comment (baurine) [inline ui/src/apps/diagnose/components/DiagnoseGenerator.tsx]: We can `required` props so the asterisk won't miss.

File: ui/src/apps/diagnose/components/DiagnoseGenerator.tsx [hunk 2]
```diff
@@ -52,18 +64,49 @@ export default function DiagnoseGenerator() {
...
+              rules={[{ required: true }]}
+              noStyle
+            >
+              <Select style={{ width: 120 }}>
+                <Select.Option value={5}>5 min</Select.Option>
+                <Select.Option value={10}>10 min</Select.Option>
+                <Select.Option value={30}>30 min</Select.Option>
+                <Select.Option value={60}>1 hour</Select.Option>
+                <Select.Option value={24 * 60}>1 day</Select.Option>
+                <Select.Option value={0}>Custom</Select.Option>
```
Comment (baurine) [inline ui/src/apps/diagnose/components/DiagnoseGenerator.tsx]: Can we support i18n here?

**Comment by baurine** [comment]
the rest LGTM!

File: ui/src/apps/diagnose/components/DiagnoseGenerator.tsx [hunk 3]
```diff
@@ -52,18 +64,49 @@ export default function DiagnoseGenerator() {
...
+                  getFieldValue('rangeDuration') === 0 && (
+                    <Form.Item
+                      noStyle
+                      name="rangeDurationCustom"
+                      rules={[{ required: true }]}
+                    >
+                      <InputNumber
+                        min={1}
+                        max={43200}
+                        formatter={(value) => `${value} min`}
```
Comment (baurine) [inline ui/src/apps/diagnose/components/DiagnoseGenerator.tsx]: Can we support i18n here as well?

**Comment by baurine** [comment]
btw, I think maybe we should change `Range Size` to `Range Duration`, `Range Size` to represent the time length sounds weird...

File: ui/src/apps/diagnose/components/DiagnoseGenerator.tsx (hunk 2 above)
Comment (breeswish) [reply ui/src/apps/diagnose/components/DiagnoseGenerator.tsx]: I guess it should be 5min (consider other abbreviations 5km, 5kg, etc)

File: ui/src/apps/diagnose/components/DiagnoseGenerator.tsx (hunk 1 above)
Comment (breeswish) [reply ui/src/apps/diagnose/components/DiagnoseGenerator.tsx]: It doesn't work. The star does not appear. Maybe it is because there is no values inside.

**Comment by breeswish** [comment]
> btw, I think maybe we should change `Range Size` to `Range Duration`, `Range Size` to represent the time length sounds weird...

I don't like "Size" as well, however "duration" will give false information that it will diagnose for this long time.

File: ui/src/apps/diagnose/components/DiagnoseGenerator.tsx (hunk 1 above)
Comment (baurine) [reply ui/src/apps/diagnose/components/DiagnoseGenerator.tsx]: I mean to use `required` as props, likes `<Form.Item required .../>`, I tested and it works:

image

File: ui/src/apps/diagnose/components/DiagnoseGenerator.tsx (hunk 2 above)
Comment (baurine) [reply ui/src/apps/diagnose/components/DiagnoseGenerator.tsx]: er, make sense.

Comment (baurine) [reply ui/src/apps/diagnose/components/DiagnoseGenerator.tsx]: Let me add i18n support soon.

**Comment by baurine** [comment]
image

image

**Pull request merged**